File: Assignment 01/Program 03/prog3.py
# Write a C program to find the minimum element in an array using dynamic memory allocation.

import array as arr

a = arr.array('i')

n = int(input("Enter the number of elements: "))
for i in range(n):
    val = int(input(f"Enter the element {i+1}: " ))
    a.append(val)

min = 1000000000
for i in range(n):
    if(min> a[i]):
        min =a[i]

print("The minimum element is",min)

File: Data Structure and Algorithm/Assignment 01/Program 03/prog3.c
// Write a C program to find the minimum element in an array using dynamic memory allocation.

#include<stdio.h>
#include<stdlib.h>
#include<limits.h>

int main(){
    int n;
    printf("Enter the number of inputs: ");
    scanf("%d",&n);
    int *ptr;
    ptr = (int *)malloc(n*sizeof(int));
    for (int i = 0; i < n; i++)
    {
        printf("Enter the element %d: ",i+1);
        scanf("%d",(ptr+i));
    }
    
    int min = INT_MAX;
    for (int i = 0; i < n; i++)
    {
        if(min> ptr[i])
            min = ptr[i];
    }

    printf("The minimum element is: %d",min);
    free(ptr);
    return 0;
}